app/src/main/java/io/appium/uiautomator2/model/settings/MjpegScalingFactor.java

```diff
@@ -39,6 +42,9 @@ public Integer getValue() {
 
     @Override
     protected void apply(Integer value) {
+        if (value == null || value < 1 || value > 100) {
+            throw new UiAutomator2Exception(String.format("Invalid %s value specified", SETTING_NAME));
```

Comment:
Please also include the description of the expected value ranges into the error message. Something like:

```
throw new IllegalArgumentException(String.format("The scaling factor value must be in range 1..100. %s was given instead", value));
```

Reply:
same below

Reply:
This will be resolved w/ the upcoming update(s)